### PR TITLE
Create support bundle spec in a secret

### PR DESCRIPTION
**Pull request by divolgin**

### CODE REVIEW

File: kotsadm/pkg/apiserver/server.go [hunk 1]
```diff
@@ -73,8 +73,6 @@ func Start() {
 	r.HandleFunc("/api/v1/logout", handlers.Logout) // this route uses its own auth
 	r.Path("/api/v1/metadata").Methods("GET").HandlerFunc(handlers.Metadata)
 
-	r.Path("/api/v1/troubleshoot").Methods("GET").HandlerFunc(handlers.GetDefaultTroubleshoot)
```
Comment (divolgin) [inline kotsadm/pkg/apiserver/server.go:76]: I could not find any references to this endpoint.

File: kotsadm/pkg/airgap/airgap.go [hunk 2]
```diff
@@ -229,6 +230,11 @@ func CreateAppFromAirgap(pendingApp *types.PendingApp, airgapPath string, regist
 		return errors.Wrap(err, "failed to load kotskinds from path")
 	}
 
+	err = supportbundle.CreateRenderedSpec(a.ID, a.CurrentSequence, "", true, kotsKinds.SupportBundle)
```
Comment (salahalsaleh) [inline kotsadm/pkg/airgap/airgap.go:233]: just want to note that `a.CurrentSequence` is _not_ the deployed sequence which is probably what you're looking for here.

File: kotsadm/pkg/socketservice/socketservice.go [hunk 3]
```diff
@@ -364,8 +364,29 @@ func (s *SocketService) processSupportBundle(clusterSocket ClusterSocket, pendin
 		return errors.Wrap(err, "failed to get socket channel from server")
 	}
 
+	currentVersion, err := downstream.GetCurrentVersion(a.ID, clusterSocket.ClusterID)
```
Comment (salahalsaleh) [inline kotsadm/pkg/socketservice/socketservice.go:367]: `currentVersion` can be nil

File: kotsadm/pkg/airgap/airgap.go (hunk 2 above)
Comment (salahalsaleh) [inline kotsadm/pkg/airgap/airgap.go:233]: is `a.CurrentSequence` the right sequence to pass here? just curious.

File: kotsadm/web/src/components/troubleshoot/GenerateSupportBundle.jsx [hunk 4]
```diff
@@ -283,7 +300,7 @@ class GenerateSupportBundle extends React.Component {
                 :
                 <div>
                   <div className="u-marginTop--40">
-                    If you'd prefer, <a href="#" className="replicated-link" onClick={(e) => this.toggleShow(e, "showRunCommand")}>click here</a> to get a command to manually generate a support bundle.
+                    If you'd prefer, <a href="#" className="replicated-link" onClick={(e) => this.fetchSupportBunldeCommand()}>click here</a> to get a command to manually generate a support bundle.
```
Comment (marccampbell) [inline kotsadm/web/src/components/troubleshoot/GenerateSupportBundle.jsx:303]: not a blocker, but that is a weird spelling of bundle

**Review by marccampbell — APPROVED** [review]

File: kotsadm/pkg/socketservice/socketservice.go (hunk 3 above)
Comment (divolgin) [reply kotsadm/pkg/socketservice/socketservice.go:367]: indeed!

**Pull request merged**